dowhy/causal_refuters/dummy_outcome_refuter.py

```diff
@@ -157,29 +190,43 @@ def refute_estimate(self):
+
+                    # Check if the value of true effect has been already stored
+                    # This ensures that we calculate the causal effect only once.
+                    # We use key_train as we map data with respect to the base category of the data
                     
-            new_data = validation_df.assign(dummy_outcome=outcome_validation)
-            new_estimator = CausalEstimator.get_estimator_object(new_data, identified_estimand, self._estimate)
-            new_effect = new_estimator.estimate_effect()
-            estimates.append(new_effect.value)
+                    if key_train not in causal_effect_map:
+                        # As we currently support only one treatment
+                        causal_effect_map[key_train] = self._true_causal_effect( validation_df[ self._treatment_name[0] ] )
+                    
+                    # Add h(t) to f(W) to get the dummy outcome
+                    outcome_validation += causal_effect_map[key_train]      
+                    new_data = validation_df.assign(dummy_outcome=outcome_validation)
+                    new_estimator = CausalEstimator.get_estimator_object(new_data, identified_estimand, self._estimate)
+                    new_effect = new_estimator.estimate_effect()
+                    estimates.append(new_effect.value)
 
-        simulation_results.append(estimates)
+            simulation_results.append(estimates)
 
         # We convert to ndarray for ease in indexing
         # The data is of the form
         # sim1: cat1 cat2 ... catn
         # sim2: cat1 cat2 ... catn
         simulation_results = np.array(simulation_results)
 
-        # Note: We hardcode the estimate value to ZERO as we want to check if it falls in the distribution of the refuter
+        # Note: We would hardcode the estimate value to ZERO as we want to check if it falls in the distribution of the refuter
         # Ideally we should expect that ZERO should fall in the distribution of the effect estimates as we have severed any causal 
         # relationship between the treatment and the outcome.
-        dummy_estimator = CausalEstimate(
-                estimate = 0,
-                target_estimand =self._estimate.target_estimand,
-                realized_estimand_expr=self._estimate.realized_estimand_expr)
+        # On the other hand, if we define a causal effect h(t) then we would like to do the same, that is, find if it falls in the 
+        # distribution of the refuter.
 
         if no_estimator:
+            
+            dummy_estimator = CausalEstimate(
+                    estimate = causal_effect_map[None],
+                    target_estimand =self._estimate.target_estimand,
+                    realized_estimand_expr=self._estimate.realized_estimand_expr)
+            
             refute = CausalRefutation(
                         self._estimate.value,
```

Comment:
is self._estimate.value correct here? I think it should be replaced by the dummy_estimate object--that's what we are comparing to.